### PR TITLE
Added ability to specify parent client tenant for a child tenant

### LINKED ISSUE #29: Allow Client Tenants to be created as children of other Client Tenants

### PR DESCRIPTION
**Pull request by edfreeman**
Resolves #29

### CODE REVIEW

File: Solutions/Marain.TenantManagement.Specs/Features/TenantManagement/CreateClientTenant.feature [hunk 1]
```diff
@@ -19,3 +19,24 @@ Scenario: Create a new client tenant with a well known GUID
...
+	And I have an existing client tenant with a well known Guid '102fdab0-6d9c-4f99-bf87-ce94ad3bbae6' called 'Fabrikam'
+	When I use the tenant management service to create a new child client tenant of the 'Fabrikam' client tenant with well known Guid 'd620a845-de25-4b88-b5a1-27d6b1104f82' called 'Litware'
+	Then the tenant with Id '75b9261673c2714681f14c97bc0439fbb0da2f109c6d994fbf87ce94ad3bbae645a820d625de884bb5a127d6b1104f82' is called 'Litware'
+	And the tenant with Id '75b9261673c2714681f14c97bc0439fbb0da2f109c6d994fbf87ce94ad3bbae645a820d625de884bb5a127d6b1104f82' has a parent with Id '75b9261673c2714681f14c97bc0439fbb0da2f109c6d994fbf87ce94ad3bbae6'
+
+Scenario: Create a new child client tenant of a client tenant which does not exist 
+	Given the tenancy provider has been initialised for use with Marain
+	And the client tenant 'Fabrikam' does not exist
+	When I use the tenant management service to create a new child client tenant of the 'Fabrikam' client tenant with well known Guid 'd620a845-de25-4b88-b5a1-27d6b1104f82' called 'Litware'
+	Then an 'TenantNotFoundException' is thrown
```
Comment (jongeorge1) [inline Solutions/Marain.TenantManagement.Specs/Features/TenantManagement/CreateClientTenant.feature:34]: Super-picky: this should be *a*, not *an*.

File: Solutions/Marain.TenantManagement.Specs/Steps/TenantSteps.cs [hunk 2]
```diff
@@ -40,18 +40,25 @@ public Task WhenIUseTheTenantManagementServiceToCreateANewClientTenantCalled(str
                 });
         }
 
+        [Given(@"I have an existing client tenant with a well known Guid '(.*)' called '(.*)'")]
```
Comment (jongeorge1) [inline Solutions/Marain.TenantManagement.Specs/Steps/TenantSteps.cs:43]: We normally try and remove the `@` if they aren't needed. Roslynator should be able to fix all of these for you in one go.

File: Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs [hunk 3]
```diff
@@ -27,16 +27,21 @@ public CreateClientTenantCommand(ITenantManagementService tenantManagementServic
                 Description = "The name of the new client.",
                 Arity = ArgumentArity.ExactlyOne,
             };
+            var parentId = new Option<string>("--parentId")
+            {
+                Description = "[Optional] The ID of the parent Client Tenant.",
```
Comment (jongeorge1) [inline Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs:32]: You probably don't need `[Optional]` in here - my understanding is that options are all optional - when you get the help, they end up looking like this:

```
Usage:
  marain enroll [options] <clientTenantId> <serviceTenantId>

Arguments:
  <clientTenantId>     The Id of the client tenant.
  <serviceTenantId>    The Id of the service tenant.

Options:
  --config <config>    JSON configuration file to use when enrolling.
  -?, -h, --help       Show help and usage information
```

File: Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs [hunk 4]
```diff
@@ -27,16 +27,21 @@ public CreateClientTenantCommand(ITenantManagementService tenantManagementServic
...
+            };
 
             this.AddArgument(clientName);
+            this.AddOption(parentId);
 
-            this.Handler = CommandHandler.Create((string name) => this.HandleCommand(name));
+            this.Handler = CommandHandler.Create((string name, string parentId) => this.HandleCommand(name, parentId));
```
Comment (jongeorge1) [inline Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs:38]: Does `System.CommandLine` not like it if you use `string? parentId` here? I know it does some magic to map the params to the arguments/options, but I don't know how sensitive it is...

File: Solutions/Marain.TenantManagement.Specs/Features/TenantManagement/CreateClientTenant.feature (hunk 1 above)
Comment (edfreeman) [reply Solutions/Marain.TenantManagement.Specs/Features/TenantManagement/CreateClientTenant.feature]: Copy-paste fail. Fixed

File: Solutions/Marain.TenantManagement.Specs/Steps/TenantSteps.cs (hunk 2 above)
Comment (edfreeman) [reply Solutions/Marain.TenantManagement.Specs/Steps/TenantSteps.cs]: Removed

File: Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs (hunk 3 above)
Comment (edfreeman) [reply Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs]: Removed

File: Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs (hunk 4 above)
Comment (edfreeman) [reply Solutions/Marain.TenantManagement.Cli/Marain/TenantManagement/Cli/Commands/CreateClientTenantCommand.cs]: Seems to be happy - changed

**Pull request merged**